include/slowly.h

```diff
@@ -0,0 +1,31 @@
+#ifndef SLOWLY_H
+#define SLOWLY_H
+
+#include "ultra64.h"
+
+#define SLOWLY_STATUS_DONE (1 << 0)
+#define SLOWLY_STATUS_STARTED (1 << 1)
+
+typedef void (*SlowlyCallbackZero)(void);
+typedef void (*SlowlyCallbackOne)(void*);
+typedef void (*SlowlyCallbackTwo)(void*, void*);
+
+typedef union {
+    SlowlyCallbackZero zero;
+    SlowlyCallbackOne one;
+    SlowlyCallbackTwo two;
+} SlowlyCallback; // size = 0x4
```

Comment:
Even with `()` and no union still getting typed warnings from IDO about anonymous functions when actually calling it (but surprisingly GCC seems fine with it 🤔). I'm also pretty sure this syntax is being phased out in general so would prefer to avoid it.

Reply:
The whole thing's pretty unsafe anyway, it assumes that the function has two arguments when it initialises it, doesn't it? Possibly it could be overridden later.

Reply:
GCC's silence is surprising though.